### PR TITLE
Add support and release content to CAPIBM book doc

### PR DESCRIPTION
**Pull request by Niharika0306**
**What this PR does / why we need it**:
This PR adds the support and release content to CAPIBM book doc.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #[1498](https://github.com/kubernetes-sigs/cluster-api-provider-ibmcloud/issues/1498)

**Special notes for your reviewer**:

/area provider/ibmcloud

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note
 Add support and release content to CAPIBM book doc
```

### CODE REVIEW

File: docs/book/src/SUMMARY.md [hunk 1]
```diff
@@ -31,6 +31,7 @@
   - [Rapid iterative development with Tilt](./developer/tilt.md)
   - [Guide for API conversions](./developer/conversion.md)
   - [Release Process](./developer/release.md)
+  - [Contributing Guidelines](./developer/contributing-guidelines.md)
```
Comment (Prajyot-Parab) [inline docs/book/src/SUMMARY.md:34]:
```suggestion
  - [Release Support Guidelines](./developer/release-support-guidelines.md)
```

@Niharika0306 Lets rename this as above.

File: docs/book/src/developer/contributing-guidelines.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Contributing guidelines
```
Comment (Prajyot-Parab) [inline docs/book/src/developer/contributing-guidelines.md:1]:
```suggestion
# Release Support Guidelines
```

File: docs/book/src/developer/index.md [hunk 3]
```diff
@@ -3,6 +3,7 @@
 - [Rapid iterative development with Tilt](/developer/tilt.html)
 - [Guide for API conversions](/developer/conversion.html)
 - [Release Process](/developer/release.html)
+- [Contributing Guidelines](/developer/contributing-guidelines.md)
```
Comment (Prajyot-Parab) [inline docs/book/src/developer/index.md:6]:
```suggestion
- [Release Support Guidelines](/developer/release-support-guidelines.md)
```

File: docs/book/src/developer/contributing-guidelines.md
Comment (Prajyot-Parab) [inline docs/book/src/developer/contributing-guidelines.md:1]: Rename file - `release-support-guidelines.md`

**Review by Prajyot-Parab — CHANGES_REQUESTED** [review]

File: docs/book/src/SUMMARY.md (hunk 1 above)
Comment (Niharika0306) [reply docs/book/src/SUMMARY.md]: Yes, makes sense. I have addressed it.

**Comment by Prajyot-Parab** [comment]
ptal @mkumatag

**Review by mkumatag — APPROVED** [review]
/lgtm

**Pull request merged**